### PR TITLE
Add Go proto build rule for p4info

### PR DESCRIPTION
**Pull request by sthesayi**
This PR adds a build rule for the p4info.proto to generate the Go library version.

### CODE REVIEW

**Review by smolkaj — APPROVED** [review]
LGTM.
@antoninbas could you take a look?

**Comment by antoninbas** [comment]
The PR itself looks good to me, but the build is failing: `ERROR: /home/runner/work/p4runtime/p4runtime/proto/BUILD.bazel:63:1: name 'go_proto_library' is not defined (did you mean 'cc_proto_library'?)`

Please also take a look at the CLA

**Review by smolkaj — CHANGES_REQUESTED** [review]
Undo LGTM, look like we need additional changes to make things build.

**Comment by sthesayi** [comment]
I believe to have fixed the build issue. Sorry for the premature request as it was my first CL and I forgot to mark it as draft. I am not sure why it still says "Waiting for status to be reported" but the CL build is passing now. I also fixed the CLA issue.

**Comment by sthesayi** [comment]
> Undo LGTM, look like we need additional changes to make things build.

This is good for second review. Thanks

**Review by smolkaj — APPROVED** [review]
LGTM modulo the go.{mod, sum} changes, since I'm not a go user.
@antoninbas could you take a look.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,11 @@ module github.com/p4lang/p4runtime
...
 require (
-	github.com/golang/protobuf v1.4.0
-	google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36
+	github.com/golang/protobuf v1.4.3
+	github.com/google/go-cmp v0.5.1 // indirect
+	golang.org/x/sys v0.0.0-20200930185726-fdedc70b468f // indirect
+	golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1 // indirect
+	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013
 	google.golang.org/grpc v1.28.1
+	google.golang.org/protobuf v1.25.0 // indirect
```
Comment (antoninbas) [inline go.mod:12]: I am also unsure about the go.mod changes. I tried a few things (e.g., removing all required modules and running `go mod tidy`) but I could never get these indirect dependencies. Could you explain how they got there?

Comment (sthesayi) [reply go.mod:12]: I ran the codegen/update.sh. Without that the build was failing because the Go modules were not up to date.

Comment (sthesayi) [reply go.mod:12]: Also some of these might be due to Google auto updater which is defaulted in my emacs. Is there an issue with keeping all of these dependencies up-to-date? I can always try to handcraft the go.mod to remove the indirect ones and see if it is still ok with the build but I am a bit afraid messing with this by hand.

**Review by antoninbas — APPROVED** [review]

**Pull request merged**